octobot/src/server/github_handler.rs

```diff
@@ -409,22 +411,34 @@ impl GithubEventHandler {
         participants.push(pull_request.user().clone());
         // add team members
         let teams = pull_request.teams();
-        let futures = teams
-            .iter()
-            .map(|t| self.github_session.get_team_members(&t.url));
-        let team_members = future::join_all(futures).await;
-        participants.extend(
-            team_members
-                .into_iter()
-                .filter_map(|r| match r {
-                    Ok(r) => Some(r),
+        for t in teams {
+            if self
+                .team_members_cache
+                .lock()
+                .unwrap()
+                .contains_key(&t.slug)
+            {
+                let cache = self.team_members_cache.lock().unwrap();
```

Comment:
note: releasing the lock and grabbing it again like this can results in thread-safety issues. 

1. thread one checks for existence -- `contains_key` returns true
2. thread two grabs from the ttl cache and the time expires, removes it
3. thread 1 tries to grab from the ttl cache and it's gone 

better to keep the lock held both for the `contains` and the `lookup`.

or, better still: just call `get`, and use the `Option` it returns to decided whether it was successful or not.